### ISSUE TITLE
Pretty printer occasionally raises TypeError when displaying vectors from the mechanics module on Jupyter Notebook

### ISSUE THREAD

**Issue by Davide-sd** [sympy/sympy#26731]
I'm running this on Jupyter Notebook v7.1.3.

sympy: 1.13.0rc2

Example code:

```py
from sympy import *
from sympy.physics.mechanics import *
init_vprinting()

q1, q2 = dynamicsymbols("q1, q2")
r = symbols("r")

L, C, R = symbols("L, C, R", cls=ReferenceFrame)
C.orient_axis(L, L.x, q1)
R.orient_axis(C, C.y, q2)

P = r*C.y - 3*r * R.x
```

This works fine:

```py
C.y.diff(q1, L).express(C).simplify()
```
![image](https://github.com/sympy/sympy/assets/9921777/15144d19-7144-4407-a9c3-3625206d7cf1)

This raises TypeError:

```py
C.z.diff(q1, L).express(C).simplify()
```
```text
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File [~/Documents/Development/envs/plot/lib/python3.12/site-packages/IPython/core/formatters.py:711](http://localhost:8888/envs/plot/lib/python3.12/site-packages/IPython/core/formatters.py#line=710), in PlainTextFormatter.__call__(self, obj)
    704 stream = StringIO()
    705 printer = pretty.RepresentationPrinter(stream, self.verbose,
    706     self.max_width, self.newline,
    707     max_seq_length=self.max_seq_length,
    708     singleton_pprinters=self.singleton_printers,
    709     type_pprinters=self.type_printers,
    710     deferred_pprinters=self.deferred_printers)
--> 711 printer.pretty(obj)
    712 printer.flush()
    713 return stream.getvalue()

File [~/Documents/Development/envs/plot/lib/python3.12/site-packages/IPython/lib/pretty.py:394](http://localhost:8888/envs/plot/lib/python3.12/site-packages/IPython/lib/pretty.py#line=393), in RepresentationPrinter.pretty(self, obj)
    391 for cls in _get_mro(obj_class):
    392     if cls in self.type_pprinters:
    393         # printer registered in self.type_pprinters
--> 394         return self.type_pprinters[cls](obj, self, cycle)
    395     else:
    396         # deferred printer
    397         printer = self._in_deferred_types(cls)

File [~/Documents/Development/envs/plot/lib/python3.12/site-packages/sympy/interactive/printing.py:82](http://localhost:8888/envs/plot/lib/python3.12/site-packages/sympy/interactive/printing.py#line=81), in _init_ipython_printing.<locals>._print_plain(arg, p, cycle)
     80 """caller for pretty, for use in IPython 0.11"""
     81 if _can_print(arg):
---> 82     p.text(stringify_func(arg))
     83 else:
     84     p.text(IPython.lib.pretty.pretty(arg))

File [~/Documents/Development/envs/plot/lib/python3.12/site-packages/sympy/physics/vector/printing.py:285](http://localhost:8888/envs/plot/lib/python3.12/site-packages/sympy/physics/vector/printing.py#line=284), in vpprint(expr, **settings)
    282 uflag = pretty_use_unicode(use_unicode)
    284 try:
--> 285     return pp.doprint(expr)
    286 finally:
    287     pretty_use_unicode(uflag)

File [~/Documents/Development/envs/plot/lib/python3.12/site-packages/sympy/printing/pretty/pretty.py:66](http://localhost:8888/envs/plot/lib/python3.12/site-packages/sympy/printing/pretty/pretty.py#line=65), in PrettyPrinter.doprint(self, expr)
     65 def doprint(self, expr):
---> 66     return self._print(expr).render(**self._settings)

File [~/Documents/Development/envs/plot/lib/python3.12/site-packages/sympy/printing/printer.py:309](http://localhost:8888/envs/plot/lib/python3.12/site-packages/sympy/printing/printer.py#line=308), in Printer._print(self, expr, **kwargs)
    307 if self.printmethod and hasattr(expr, self.printmethod):
    308     if not (isinstance(expr, type) and issubclass(expr, Basic)):
--> 309         return getattr(expr, self.printmethod)(self, **kwargs)
    311 # See if the class of expr is known, or if one of its super
    312 # classes is known, and use that print function
    313 # Exception: ignore the subclasses of Undefined, so that, e.g.,
    314 # Function('gamma') does not get dispatched to _print_gamma
    315 classes = type(expr).__mro__

File [~/Documents/Development/envs/plot/lib/python3.12/site-packages/sympy/physics/vector/vector.py:278](http://localhost:8888/envs/plot/lib/python3.12/site-packages/sympy/physics/vector/vector.py#line=277), in Vector._pretty(self, printer)
    276     terms.append(prettyForm(N.pretty_vecs[i]))
    277 elif M[i] == -1:
--> 278     terms.append(prettyForm.NEG + prettyForm(N.pretty_vecs[i]))
    279 else:
    280     terms.append(juxtapose(M[i], N.pretty_vecs[i]))

TypeError: unsupported operand type(s) for +: 'int' and 'prettyForm'
```

But somehow still manages to display the correct value:
![image](https://github.com/sympy/sympy/assets/9921777/bd2f163b-3993-4a1a-880c-94308a1ec1cf)

**Comment by moorepants** [comment]
Both the latex and pretty printers are run in the background, so the error is from the pretty printer, but the successful latex printed form displays.

**Comment by moorepants** [comment]
In IPython:
```python
$ ipython
Python 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:45:18) [GCC 12.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.25.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %paste
from sympy import *
from sympy.physics.mechanics import *
init_vprinting()

q1, q2 = dynamicsymbols("q1, q2")
r = symbols("r")

L, C, R = symbols("L, C, R", cls=ReferenceFrame)
C.orient_axis(L, L.x, q1)
R.orient_axis(C, C.y, q2)

P = r*C.y - 3*r * R.x

## -- End pasted text --

In [2]: C.y.diff(q1, L).express(C).simplify()
Out[2]: c_z

In [3]: C.z.diff(q1, L).express(C).simplify()
Out[3]: ---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/miniconda/lib/python3.10/site-packages/IPython/core/formatters.py:711, in PlainTextFormatter.__call__(self, obj)
    704 stream = StringIO()
    705 printer = pretty.RepresentationPrinter(stream, self.verbose,
    706     self.max_width, self.newline,
    707     max_seq_length=self.max_seq_length,
    708     singleton_pprinters=self.singleton_printers,
    709     type_pprinters=self.type_printers,
    710     deferred_pprinters=self.deferred_printers)
--> 711 printer.pretty(obj)
    712 printer.flush()
    713 return stream.getvalue()

File ~/miniconda/lib/python3.10/site-packages/IPython/lib/pretty.py:394, in RepresentationPrinter.pretty(self, obj)
    391 for cls in _get_mro(obj_class):
    392     if cls in self.type_pprinters:
    393         # printer registered in self.type_pprinters
--> 394         return self.type_pprinters[cls](obj, self, cycle)
    395     else:
    396         # deferred printer
    397         printer = self._in_deferred_types(cls)

File ~/src/sympy/sympy/interactive/printing.py:82, in _init_ipython_printing.<locals>._print_plain(arg, p, cycle)
     80 """caller for pretty, for use in IPython 0.11"""
     81 if _can_print(arg):
---> 82     p.text(stringify_func(arg))
     83 else:
     84     p.text(IPython.lib.pretty.pretty(arg))

File ~/src/sympy/sympy/physics/vector/printing.py:285, in vpprint(expr, **settings)
    282 uflag = pretty_use_unicode(use_unicode)
    284 try:
--> 285     return pp.doprint(expr)
    286 finally:
    287     pretty_use_unicode(uflag)

File ~/src/sympy/sympy/printing/pretty/pretty.py:66, in PrettyPrinter.doprint(self, expr)
     65 def doprint(self, expr):
---> 66     return self._print(expr).render(**self._settings)

File ~/src/sympy/sympy/printing/printer.py:309, in Printer._print(self, expr, **kwargs)
    307 if self.printmethod and hasattr(expr, self.printmethod):
    308     if not (isinstance(expr, type) and issubclass(expr, Basic)):
--> 309         return getattr(expr, self.printmethod)(self, **kwargs)
    311 # See if the class of expr is known, or if one of its super
    312 # classes is known, and use that print function
    313 # Exception: ignore the subclasses of Undefined, so that, e.g.,
    314 # Function('gamma') does not get dispatched to _print_gamma
    315 classes = type(expr).__mro__

File ~/src/sympy/sympy/physics/vector/vector.py:278, in Vector._pretty(self, printer)
    276     terms.append(prettyForm(N.pretty_vecs[i]))
    277 elif M[i] == -1:
--> 278     terms.append(prettyForm.NEG + prettyForm(N.pretty_vecs[i]))
    279 else:
    280     terms.append(juxtapose(M[i], N.pretty_vecs[i]))

TypeError: unsupported operand type(s) for +: 'int' and 'prettyForm'
```

**Comment by oscarbenjamin** [comment]
Bisected to 8935af27128f49143d4b5822232c0e1097964945 from gh-25753.

**Comment by oscarbenjamin** [comment]
Possible fix:
```diff
diff --git a/sympy/physics/vector/vector.py b/sympy/physics/vector/vector.py
index 8fd36307dc..ef9d593f49 100644
--- a/sympy/physics/vector/vector.py
+++ b/sympy/physics/vector/vector.py
@@ -275,7 +275,7 @@ def juxtapose(a, b):
                 elif M[i] == 1:
                     terms.append(prettyForm(N.pretty_vecs[i]))
                 elif M[i] == -1:
-                    terms.append(prettyForm.NEG + prettyForm(N.pretty_vecs[i]))
+                    terms.append(prettyForm("-1") * prettyForm(N.pretty_vecs[i]))
                 else:
                     terms.append(juxtapose(M[i], N.pretty_vecs[i]))
```
Minimal reproducer:
```python
In [7]: from sympy.physics.vector import ReferenceFrame

In [8]: -ReferenceFrame('C').y
Out[8]: ---------------------------------------------------------------------------
TypeError 
```

**Comment by oscarbenjamin** [comment]
Thanks for reporting @Davide-sd

**Comment by oscarbenjamin** [comment]
@Davide-sd this should be fixed in sympy 1.13.0rc4. Let me know if it works for you.

**Comment by Davide-sd** [comment]
@oscarbenjamin It works perfectly, thank you :)
